forcetree.c

```diff
@@ -989,8 +991,8 @@ void force_update_hmax(int * activeset, int size)
     /* FIXME: actually only TOPLEVEL nodes contains the local mass can potentially be dirty,
      *  we may want to save a list of them to speed this up.
      * */
-    for(i = All.MaxPart; i < All.MaxPart + MaxNodes; i ++) {
-        NodeIsDirty[i - All.MaxPart] = 0;
+    for(i = 0; i < MaxNodes; i ++) {
+        NodeIsDirty[i] = 0;
```

Comment:
Is this the only variable per node? Did we settle on names for SingleParticleNode/TerminalNode and TreeNode? Perhaps consider start making this a struct member here?

Reply:
Yes, that one can be cleaned up substantially. I'll make a note though and defer it to a longer cleanup of the tree - because these commits split All.MaxPart from MaxNodes it enables some more tree cleanups. I also want to make the tree build more thread-scalable.